src/medCore/data/medAbstractData.cpp

```diff
@@ -43,6 +43,13 @@ medAbstractData::medAbstractData( medAbstractData *parent )
     this->moveToThread(QApplication::instance()->thread());
 }
 
+medAbstractData::medAbstractData(const medAbstractData& other)
+    : dtkAbstractData(other),
+      d(new medAbstractDataPrivate())
+{
+    *d = *other.d;
```

Comment:
Instances of ```medAbstractData``` contain a pointer to a ```medAttachedData``` object, and so that pointer needs to be cloned. I checked the classes deriving from ```medAttachedData``` and the only one that needs a proper copy constructor for cloning to work is ```medImageMaskAnnotationData```. That copy constructor will need to clone the ```m_maskData``` member (which now works since the mask type is now clonable thanks to the other work you've done on this PR).
After that I think this should be over.